grpc-server/hardware.go

```diff
@@ -297,22 +309,35 @@ func (s *server) Delete(ctx context.Context, in *hardware.DeleteRequest) (*hardw
 }
 
 func (s *server) validateHardwareData(ctx context.Context, hw *hardware.Hardware) error {
-	interfaces := hw.GetNetwork().GetInterfaces()
-	for i := range hw.GetNetwork().GetInterfaces() {
-		data, _ := s.db.GetByMAC(ctx, interfaces[i].GetDhcp().GetMac())
-		if data != "" {
-			logger.With("MAC", interfaces[i].GetDhcp().GetMac()).Info(duplicateMAC)
+	for _, iface := range hw.GetNetwork().GetInterfaces() {
+		mac := iface.GetDhcp().GetMac()
```

Comment:
I'm wondering if it would make sense to add a test for the empty string here prior to calling `s.db.GetByMAC()`.

Would we expect that to be an error condition or would we allow a hardware definition without a mac address?

Reply:
I was questioning similar, whether we should transform the input at all, just let validate/push throw an error if the hardware mac was not "valid" (lowercase, non-empty).

That said, I don't know hardware address can be treated as optional.

Reply:
I worry that transforming the data means that any stateful tools that read back the lowercase value will need to be aware of the transform and copy that logic themselves.

Reply:
The alternative I thought of was to make the comparisons case-insensitive, but with it being stored embedded inside of a stored json blob that seemed a bit more error prone.